### DISCUSSION TITLE
How to modify and add a manual sync feature?

### DISCUSSION THREAD

**Discussion by canonbrother** [paritytech/substrate#13060] [Q&A]
Substrate blockchain automatically syncs up upon starting, however, I'd like to have manual sync feature for some devs purpose.

```mermaid
flowchart LR

subgraph P2P Network

%% n2 -- 1. discover --> n1
%% n2 -- 2. connect --> n1
n2 -- rpc::sync --> n1

subgraph n1
n1.1 --> n1.2
end

subgraph n2
n2.1 -.-> n2.2
end
end
```

#### Todo
- [ ] Add manual sync mode to stop auto sync by default
- [ ] Add `rpc::sync(Option<Hash>)` to start sync (discover -> connect) node, stop sync on target hash while `Option<Hash>` is provided else on best

As I am new to substrate and rust, this is not quite as straightforward as I would have thought, I simply cannot figure out how to proceed.

**Comment by bkchr** [comment]
This is currently not that easy possible. After [this issue](https://github.com/paritytech/substrate/issues/8686) is done, it should maybe be a little bit easier. 

However, could you describe on what you want to achieve with this?
